semeio/jobs/auto_correlate_observations/config.py

```diff
@@ -0,0 +1,315 @@
+
+    @cs.validator_msg(msg)
+    def validator(value):
+        return value in valid_values
+
+    return validator
+
+
+@cs.validator_msg("t must be an integer if a max cluster criteria is choosen")
+def _t_is_int_if_maxclust(spearman_config):
+    if spearman_config[_CRITERION] in (_MAXCLUST, _MAXCLUST_MONOCRIT):
+        return isinstance(spearman_config[_T], int)
+    return True
+
+
+@cs.transformation_msg(
+    (
+        "Will be defaulted to 5 for all max cluster criterias and "
+        "1.15 for all other criterias"
+    )
+)
+def _inject_default_t(config):
+    config = copy.deepcopy(config)
+    criterion = config.get(_CRITERION)
+    if _T in config:
+        pass
+    elif criterion in (_MAXCLUST, _MAXCLUST_MONOCRIT):
+        config[_T] = 5
+    elif criterion in (_INCONSISTENT, _DISTANCE, _MONOCRIT):
+        config[_T] = 1.15
+    else:
+        raise ValueError("Unknown criterion: {}".format(criterion))
+    return config
+
+
+_SPEARMAN_CORRELATION_SCHEMA = {
+    cs.MetaKeys.Type: cs.types.NamedDict,
+    cs.MetaKeys.Description: "TODO",
+    cs.MetaKeys.Content: {
+        _FCLUSTER: {
+            cs.MetaKeys.Type: cs.types.NamedDict,
+            cs.MetaKeys.ElementValidators: (_t_is_int_if_maxclust,),
+            cs.MetaKeys.Transformation: _inject_default_t,
+            cs.MetaKeys.Content: {
+                _T: {
+                    cs.MetaKeys.Type: cs.types.Number,
+                    cs.MetaKeys.ElementValidators: (
+                        _bounds_validator(lower=0, lower_inclusive=False),
+                    ),
+                },
+                _CRITERION: {
+                    cs.MetaKeys.Type: cs.types.String,
+                    cs.MetaKeys.ElementValidators: (
+                        _one_of(
+                            _INCONSISTENT,
+                            _DISTANCE,
+                            _MAXCLUST,
+                            _MONOCRIT,
+                            _MAXCLUST_MONOCRIT,
+                        ),
+                    ),
+                },
+                _DEPTH: {
+                    cs.MetaKeys.Type: cs.types.Integer,
+                    cs.MetaKeys.ElementValidators: (_bounds_validator(lower=1),),
+                },
+            },
+        },
+        _LINKAGE: {
+            cs.MetaKeys.Type: cs.types.NamedDict,
+            cs.MetaKeys.Content: {
+                _METHOD: {
+                    cs.MetaKeys.Type: cs.types.String,
+                    cs.MetaKeys.ElementValidators: (
+                        _one_of(
+                            _SINGLE, _COMPLETE, _AVERAGE, _WEIGHTED, _CENTROID, _WARD,
+                        ),
+                    ),
+                },
+                _METRIC: {
+                    cs.MetaKeys.Type: cs.types.String,
+                    cs.MetaKeys.ElementValidators: (_one_of(*_METRICS),),
+                },
+            },
+        },
+    },
+}
+
+
+_CLUSTERING_SCHEMA = {
```

Comment:
Is the separation of `_CLUSTERING_SCHEMA` and `_SPEARMAN_CORRELATION_SCHEMA` due to readability or something else? Also why the name `_SPEARMAN_CORRELATION_SCHEMA` when it deals only with `fcluster` in scipy?

Reply:
The separation is purely for readability and specifically to avoid 10 indents in a dictionary. I named it `_SPEARMAN_CORRELATION_SCHEMA` because it represent all the parameters that can be configured for our clustering algorithm based on spearman correlation. Happy to change the name if you have a suggestion :)

Reply:
Lazy to bring up a good name besides `OBSERVATION_CLUSTERING` so we leave at that :)